patches/server/0887-Restore-vanilla-entity-drops-behavior.patch

```diff
@@ -0,0 +1,177 @@
+                 this.armorItems.set(i, ItemStack.EMPTY);
+             }
+         }
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index a3c8a0291fa9ae6f3c96d937dd4621edd7c48535..a36d6f0e626d0e0084556eb191b1fa65e806c829 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -862,17 +862,17 @@ public class CraftEventFactory {
+     }
+ 
+     public static EntityDeathEvent callEntityDeathEvent(net.minecraft.world.entity.LivingEntity victim) {
+-        return CraftEventFactory.callEntityDeathEvent(victim, new ArrayList<org.bukkit.inventory.ItemStack>(0));
++        return CraftEventFactory.callEntityDeathEvent(victim, Collections.emptyMap()); // Paper
+     }
+ 
+-    public static EntityDeathEvent callEntityDeathEvent(net.minecraft.world.entity.LivingEntity victim, List<org.bukkit.inventory.ItemStack> drops) {
++    public static EntityDeathEvent callEntityDeathEvent(net.minecraft.world.entity.LivingEntity victim, Map<org.bukkit.inventory.ItemStack, java.util.function.Consumer<ItemStack>> drops) { // Paper
+         // Paper start
+         return CraftEventFactory.callEntityDeathEvent(victim, drops, com.google.common.util.concurrent.Runnables.doNothing());
+     }
+-    public static EntityDeathEvent callEntityDeathEvent(net.minecraft.world.entity.LivingEntity victim, List<org.bukkit.inventory.ItemStack> drops, Runnable lootCheck) {
++    public static EntityDeathEvent callEntityDeathEvent(net.minecraft.world.entity.LivingEntity victim, Map<org.bukkit.inventory.ItemStack, java.util.function.Consumer<ItemStack>> drops, Runnable lootCheck) { // Paper
+         // Paper end
+         CraftLivingEntity entity = (CraftLivingEntity) victim.getBukkitEntity();
+-        EntityDeathEvent event = new EntityDeathEvent(entity, drops, victim.getExpReward());
++        EntityDeathEvent event = new EntityDeathEvent(entity, new ArrayList<>(drops.keySet()), victim.getExpReward()); // Paper
+         populateFields(victim, event); // Paper - make cancellable
+         CraftWorld world = (CraftWorld) entity.getWorld();
+         Bukkit.getServer().getPluginManager().callEvent(event);
+@@ -889,16 +889,22 @@ public class CraftEventFactory {
+         for (org.bukkit.inventory.ItemStack stack : event.getDrops()) {
+             if (stack == null || stack.getType() == Material.AIR || stack.getAmount() == 0) continue;
+ 
+-            world.dropItem(entity.getLocation(), stack); // Paper - note: dropItem already clones due to this being bukkit -> NMS
++            // Paper start
++            if (drops.containsKey(stack)) {
++                drops.get(stack).accept(CraftItemStack.asNMSCopy(stack));
```

Comment:
```suggestion
+            // Paper start
+            var consumer = drops.get(stack);
+            if (consumer != null) {
+                consumer.accept(CraftItemStack.asNMSCopy(stack));
```
Consider caching `Map#get` instead of using `containsKey` and `get`. Micro optimization but worth than nothing.